fahrschein/src/main/java/org/zalando/fahrschein/ZignAccessTokenProvider.java

```diff
@@ -42,18 +43,25 @@ private static String readAll(InputStream inputStream) throws IOException {
 
     private static String zign() throws IOException {
         LOG.info("Refreshing token from zign...");
-        final Process zign = new ProcessBuilder("zign", "token", "uid").start();
+        final Process zign = new ProcessBuilder("zign", "token").start();
         try (final InputStream inputStream = zign.getInputStream()) {
-            return readAll(inputStream).trim();
-        } finally {
-            LOG.debug("Refreshed token from zign");
+            final String output = readAll(inputStream).trim();
+            zign.waitFor(5, TimeUnit.SECONDS);
+            if (zign.exitValue() != 0) {
+                throw new IOException("zign exit code was non-zero");
```

Comment:
Could include the exit code in the exception message

Reply:
Sure